lib/modules/platform/codecommit/index.ts

```diff
@@ -0,0 +1,714 @@
+import { PullRequestStatusEnum } from '@aws-sdk/client-codecommit';
+import type { Credentials } from '@aws-sdk/types';
+import JSON5 from 'json5';
+import { TextDecoder } from 'web-encoding';
+import {
+  PLATFORM_BAD_CREDENTIALS,
+  REPOSITORY_EMPTY,
+  REPOSITORY_NOT_FOUND,
+} from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { BranchStatus, PrState, VulnerabilityAlert } from '../../../types';
+import * as git from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import { sanitize } from '../../../util/sanitize';
+import type {
+  BranchStatusConfig,
+  CreatePRConfig,
+  EnsureCommentConfig,
+  EnsureCommentRemovalConfig,
+  EnsureIssueConfig,
+  EnsureIssueResult,
+  FindPRConfig,
+  Issue,
+  MergePRConfig,
+  PlatformParams,
+  PlatformResult,
+  Pr,
+  RepoParams,
+  RepoResult,
+  UpdatePrConfig,
+} from '../types';
+import { getNewBranchName, repoFingerprint } from '../util';
+import { smartTruncate } from '../utils/pr-body';
+import { getCodeCommitUrl } from './codecommit-client';
+import * as client from './codecommit-client';
+import { getUserArn, initIamClient } from './iam-client';
+
+const decoder = new TextDecoder();
```

Comment:
i still don't understand why we can't use node Buffer instead.

Reply:
i read about https://nodejs.org/api/buffer.html
i'm testing that

Reply:
ok, changed that.